### PR TITLE
fix(subtitles): auto-hide error state and harden YouTube fetcher

### PR DESCRIPTION
**Pull request by sdxdlgz**
## Type of Changes

- [x] 🐛 Bug fix (fix)

## Description

This PR improves stability of the video subtitles overlay:

- Auto-hide the `error` state after 5 seconds to avoid leaving the overlay stuck in an error UI.
- Fix a race where interceptor messages dispatched synchronously from the CC button click could be missed (set up the pending promise handlers before clicking).
- Ignore unrelated `postMessage` traffic and only process messages while a fetch is pending.
- Handle empty payloads / invalid JSON more safely and return a user-friendly “No subtitles found” error when appropriate (e.g. CC button disabled).

## Related Issue

N/A

## How Has This Been Tested?

- [x] Added unit tests
- [x] Verified through manual testing

## Checklist

- [x] I have tested these changes locally
- [ ] I have updated the documentation accordingly if necessary
- [x] My code follows the code style of this project
- [x] My changes do not break existing functionality
- [x] If my code was generated by AI, I have proofread and improved it as necessary.

---
## Summary by cubic
Improves subtitles stability by auto-hiding error states and hardening the YouTube subtitles fetcher. Prevents stuck overlays and shows clearer errors.

- **Bug Fixes**
  - Auto-hide error state after 5s to avoid a stuck overlay.
  - Set up fetch handlers before clicking the CC button to avoid missing synchronous interceptor messages.
  - Ignore unrelated postMessage traffic; only process messages while a fetch is pending and from same origin.
  - Handle empty payloads and JSON parse failures safely; show “No subtitles found” when CC is disabled.
  - Added unit tests for error auto-hide and YouTube fetcher edge cases.

<sup>Written for commit 3598fe6a6499015a7dd108af57361d5128af32bf. Summary will update on new commits.</sup>

### CODE REVIEW

File: src/utils/subtitles/fetchers/youtube/index.ts [hunk 1]
```diff
@@ -134,6 +151,15 @@ export class YoutubeSubtitlesFetcher implements SubtitlesFetcher {
       return
     }
 
+    const isDisabled = ccButton.getAttribute('aria-disabled') === 'true'
+      || ccButton.getAttribute('disabled') !== null
+      || ('disabled' in ccButton && (ccButton as HTMLButtonElement).disabled)
```
Comment (taiiiyang) [inline src/utils/subtitles/fetchers/youtube/index.ts]: 这块可以调整下 (ccButton instanceof HTMLButtonElement && ccButton.disabled)

因为 disabled 属性在每个 html element 上都有

**Review by taiiiyang — CHANGES_REQUESTED** [review]
